### DISCUSSION TITLE
Commentary from visual mode, no longer comments out from begging of lind to end of line suggestion

### DISCUSSION THREAD

**Discussion by KevinBaileyCrum** [JetBrains/ideavim#534] [General]
just upgraded, so happy the :q but was fixed along with adding count to gcc command.

post upgrade i noticed from visual mode, selecting lines and typing gcc, the comment block starts and end where the cursor was.  the old way i used to be able to select a visual block, type gcc then the entire lines would be marked as comment

**Comment by AlexPl292** [comment]
Could you please share some screencast? I don't really understand how this works for you.

**Comment by citizenmatt** [comment]
I suspect this is due to a change in the handling of character wise visual mode selection to make things more consistent with `gc{motion}`. For example, `gcw` would block comment out a word in languages that support a word, and `gcaa` would comment out a method argument when the `argtextobj` extension was enabled. However, `vwgc` or `vaagc` would comment out a whole line, not the selected region. As part of the updates to commentary, character wise visual mode behaves like the motions, and you get block commenting. If you want to get line commenting, use linewise visual mode, such as `Vgc`.

**Comment by KevinBaileyCrum** [comment]
ok i  see,

so if i go to visual mode and type gcc, i get block comments :(

but 

if i enter visual mode with a capital V, then type gcc, i get multiple single line comments :)

I can work with this for now. Any ideas if there are convenient ways to make it work with little v visual mode?

**Reply by AlexPl292** [reply]
This sounds weird, could you please attach `~/.ideavimrc` and log (https://intellij-support.jetbrains.com/hc/en-us/articles/207241085-Locating-IDE-log-files) ?

**Comment by KevinBaileyCrum** [comment]
"" source your .vimrc
"source ~/.vimrc

"" -- Suggested options --
" Show a few lines of context around the cursor. Note that this makes the
" text scroll if you mouse-click near the start or end of the window.
set scrolloff=5

set visualbell
"" set noerrorbells

" Do incremental searching.
set incsearch

" Don't use Ex mode, use Q for formatting.
map Q gq


"" -- Map IDE actions to IdeaVim -- https://jb.gg/abva4t
"" Map \r to the Reformat Code action
"map \r <Action>(ReformatCode)

"" Map <leader>d to start debug
"map <leader>d <Action>(Debug)

"" Map \b to toggle the breakpoint on the current line
"map \b <Action>(ToggleLineBreakpoint)


" Find more examples here: https://jb.gg/share-ideavimrc
"" Source your .vimrc
"source ~/.vimrc

"" -- Suggested options --
" Show a few lines of context around the cursor. Note that this makes the
" text scroll if you mouse-click near the start or end of the window.

let mapleader = ","
map <space> /

" Don't use Ex mode, use Q for formatting.
map Q gq

" Smart way to move between windows
map <C-j> <C-W>j
map <C-k> <C-W>k
map <C-h> <C-W>h
map <C-l> <C-W>l

" remap finding class"
nmap ,c :action GotoClass<CR>
nmap gi :action GotoImplementation<CR>

set ignorecase
set smartcase

map <leader>tc :tabclose<cr>
map <leader>t  :tabnext<cr>
map <leader>r  :tabprevious<cr>
map <leader>k  :tabonly<cr>

inoremap ii <esc>
inoremap II <esc>

" relative line number
" set relativenumber
" regular line number"
" set number

" plugins"
set commentary

" go to next and prev err"
" inoremap <leader>m :action GotoNextError<CR>
" inoremap <leader>M :action GotoPreviousError<CR>

set scrolloff=3

" recent files"
"map <C-f>! :action RecentFiles<CR>

" highlights search results"
set hls

" commit command
:command! Com :action CheckinProject
